app/controllers/communicarts_controller.rb

```diff
@@ -52,6 +55,9 @@ def approval_response
 private
 
   def validate_access
+    request_from_email = params['email_delivery'].presence || true
+    return unless request_from_email == true
```

Comment:
Yeah, it is confusing. From the web requests, I'm using a queryparam `email_delivery=false` which bypasses the token validation.